pkg/kubelet/cm/dra/plugin/client.go

```diff
@@ -24,130 +24,181 @@ import (
 	"google.golang.org/grpc"
 	grpccodes "google.golang.org/grpc/codes"
 	grpcstatus "google.golang.org/grpc/status"
-	"k8s.io/klog/v2"
 
+	"k8s.io/klog/v2"
 	drapbv1alpha2 "k8s.io/kubelet/pkg/apis/dra/v1alpha2"
 	drapb "k8s.io/kubelet/pkg/apis/dra/v1alpha3"
 )
 
 const PluginClientTimeout = 45 * time.Second
 
-// draPluginClient encapsulates all dra plugin methods.
-type draPluginClient struct {
-	pluginName string
-	plugin     *Plugin
-}
+type (
+	nodePrepareResourceHandler   func(context.Context, *grpc.ClientConn, *plugin, *drapb.NodePrepareResourcesRequest) (*drapb.NodePrepareResourcesResponse, error)
+	nodeUnprepareResourceHandler func(context.Context, *grpc.ClientConn, *plugin, *drapb.NodeUnprepareResourcesRequest) (*drapb.NodeUnprepareResourcesResponse, error)
+)
+
+var (
+	nodePrepareResourceHandlers = map[string]nodePrepareResourceHandler{
+		"v1alpha2": v1alpha2NodePrepareResouce,
+		"v1alpha3": v1alpha3NodePrepareResouce,
+	}
+
+	nodeUnprepareResourceHandlers = map[string]nodeUnprepareResourceHandler{
+		"v1alpha2": v1alpha2NodeUnprepareResouce,
+		"v1alpha3": v1alpha3NodeUnprepareResouce,
+	}
```

Comment:
Can we simplify `plugin.go:highestSupportedVersion` using these maps?

Reply:
I can give it a try but I have multiple issues with. First those maps aren't exported and second I have another PR that aims to move the `highestSupportedVersion` to `staging/src/k8s.io/apimachinery/pkg/util/` package

Reply:
Give it a try, please. If it's hard/not feasible you'll see it.